### PR TITLE
NotFound() method on Router

### PR DESCRIPTION
**Pull request by pkieltyka**
Solves #9 and #29

### CODE REVIEW

**Comment by pkieltyka** [comment]
This implementation is alright, but it would be nice if the NotFound() handler could cascade upwards back up the chain.

ie.
```go
sub := chi.NewRouter()
sub.Get("/hello", ...)

root := chi.NewRouter()
root.Get("/", ...)
root.NotFound(func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
  w.WriteHeader(404)
  w.Write([]byte("custom 404 page"))
})
root.Mount("/subrouter", sub)
```

a 404 on root render the custom error, but in the subrouter it will be the standard http.NotFound(), unless you define it again on that router.

This should be solvable. WIP.

**Comment by pkieltyka** [comment]
technically, this PR already supports #3 because if you set http.NotFound to some handler, chi will use that by default

**Comment by pkieltyka** [comment]
This PR completes the NotFound() handler support, along with cascading up the tree looking for a not found handler.

File: mux_test.go [hunk 1]
```diff
@@ -212,6 +212,10 @@ func TestMuxPlain(t *testing.T) {
 	r.Get("/hi", func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte("bye"))
 	})
+	r.NotFound(func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(404)
+		w.Write([]byte("nothing here"))
```
Comment (c2h5oh) [inline mux_test.go]: Defaulting to https://golang.org/pkg/net/http/#NotFound would be better

**Pull request merged**